server/routes/v1/version.ts

```diff
@@ -138,3 +144,30 @@ export const updateLastViewed = [
     return res.json({ version: id, role })
   },
 ]
+
+export const deleteVersion = [
+  ensureUserRole('user'),
+  async (req: Request, res: Response) => {
+    const { id } = req.params
+    const { user } = req
+
+    const version = await findVersionById(user, id)
+
+    if (!version) {
+      throw NotFound({ code: 'version_not_found', id }, 'Unable to find version')
+    }
+
+    if (req.user.id !== version.metadata.contacts.uploader) {
+      throw Forbidden({ code: 'user_unauthorised' }, 'User is not authorised to do this operation.')
+    }
+
+    await deleteVersionRequests(version)
+    await updateDeploymentsByVersion(version, user)
+    await deleteDeploymentsByInitialVersion(version)
+    await updateModelByVersion(version)
+
+    await version.delete()
+
+    return res.json(id)
```

Comment:
Let's return `{ id }` here instead, I'd like to try to always return objects in the future.